### PR TITLE
database: query multiple tables

### PR DESCRIPTION
**Pull request by aakbarkhan**
## Description:
- Use primary key & foreign key mechanism for joining tables. Add some new tables and add foreign key columns to the existing animal's table to model one-to-many relationships. Afterward, use JOIN to query the data.
- Query multiple tables.
- Prepare complex queries that answer analytical questions as per the project requirements.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -19,3 +19,22 @@ INSERT INTO animals (id,name,date_of_birth,weight_kg,neutered,escape_attempts) V
...
+INSERT INTO species(species_id,name) VALUES (2,'Digimon');
+
+UPDATE animals
+SET species_id = (SELECT id FROM species WHERE name = 'Digimon')
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species_id = (SELECT id FROM species WHERE name = 'Pokemon')
+WHERE species_id IS NULL;
+
```
Comment (AshakaE) [inline data.sql:40]: All animals are assigned owners, here you have assigned only two animals 

- [x] Please assign the rest animals to their owners

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @aakbarkhan  🙋‍♀️, 

Good job so far! 👌
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

###  Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (aakbarkhan) [reply data.sql:40]: Done as requested.

File: schema.sql [hunk 2]
```diff
@@ -9,3 +9,27 @@ CREATE TABLE animals (
...
+    age INT,
+    PRIMARY KEY(owner_id)
+);
+
+-- species table 
+CREATE TABLE species (
+    id INT,
+    name varchar(100),
+    PRIMARY KEY(species_id)
+);
```
Comment (DammyShittu) [inline schema.sql:26]: [OPTIONAL]

- Primary keys are meant to be columns that uniquely identify a row. Foreign key links two tables together, so it is highly possible that different rows would have the same foreign key value, hence making it unfit to be the primary key. Please, go through this StackOverflow [discussion](https://stackoverflow.com/questions/10982992/is-it-fine-to-have-foreign-key-as-primary-key/10983016#:~:text=9%20Answers&text=Foreign%20keys%20are%20almost%20always,act%20as%20the%20primary%20key.) to get more information. 👍

**Review by DammyShittu — APPROVED** [review]
Hi @aakbarkhan, 

# APPROVED 🟢 

_Outstanding implementation of the features in this milestone and changes suggested by the previous reviewer.✔️_

Kudos to you 💯 

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

## CONGRATULATIONS! 🎉 

# OPTIONAL SUGGESTIONS ♻️ 

- While trying to create a table, I'd suggest you add `GENERATED ALWAYS AS IDENTITY` to the `id` column. This will make sure that a unique value is gotten for each record/row. That way, you do not need to add `id` when you're populating your table with data. Imagine you had millions of rows 😁 

Kindly check the comment below 👇 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

# Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**